magicblock-api/src/magic_validator.rs

```diff
@@ -849,6 +921,10 @@ impl MagicValidator {
         self.token.cancel();
         self.ledger_truncator.stop();
 
+        if let Some(handle) = self.claim_fees_task.take() {
```

Comment:
Once you added a `stop` method to the `claim_fees_task` just call it here without even having to know that it is an `Option`.